### ISSUE TITLE
NRE with Tarnished

### ISSUE THREAD

**Issue by acanthii** [Bubbet/BubbetsItems#11]
Tarnished seems to cause an NRE upon level change
Context: 
Modpack R2 ID: 01925670-550e-ccb2-3231-0b1ade86e5c3
Items / Equipment (upon removing tarnished and reloading the scene, it removed my Sale Star, and my Regen Scrap. Not sure why.)
![image](https://github.com/user-attachments/assets/afb0a19e-9c28-49f6-9ea3-6e0ec465fcdf)

```
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
BubbetsItems.Items.VoidLunar.Tarnished.ApplyLuck (RoR2.CharacterMaster __instance) (at <c1bf1d0f8d704185b9099239832db5c0>:IL_0033)
(wrapper dynamic-method) RoR2.CharacterMaster.DMD<RoR2.CharacterMaster::OnInventoryChanged>(RoR2.CharacterMaster)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Trampoline<RoR2.CharacterMaster::OnInventoryChanged>?663837100(RoR2.CharacterMaster)
Aetherium.Items.TierLunar.WeightedAnklet.ManageLimiter (On.RoR2.CharacterMaster+orig_OnInventoryChanged orig, RoR2.CharacterMaster self) (at <dcf4b34f89114ce0a22261d89aac2489>:IL_0001)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Hook<RoR2.CharacterMaster::OnInventoryChanged>?1072830844(RoR2.CharacterMaster)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Trampoline<RoR2.CharacterMaster::OnInventoryChanged>?-804250544(RoR2.CharacterMaster)
EnforcerPlugin.EnforcerModPlugin.CharacterMaster_OnInventoryChanged (On.RoR2.CharacterMaster+orig_OnInventoryChanged orig, RoR2.CharacterMaster self) (at <66db015f4cb246bba53a78e4a0b0ddbd>:IL_0001)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Hook<RoR2.CharacterMaster::OnInventoryChanged>?-2075596816(RoR2.CharacterMaster)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Trampoline<RoR2.CharacterMaster::OnInventoryChanged>?-1082986764(RoR2.CharacterMaster)
BulwarksHaunt.GhostWave.WaveModifiersCharacterMasterInventoryChanged (On.RoR2.CharacterMaster+orig_OnInventoryChanged orig, RoR2.CharacterMaster self) (at <204e15b6305c458092798cdee9a3b251>:IL_0001)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Hook<RoR2.CharacterMaster::OnInventoryChanged>?83153676(RoR2.CharacterMaster)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Trampoline<RoR2.CharacterMaster::OnInventoryChanged>?-1472661566(RoR2.CharacterMaster)
MSU.BaseItemMasterBehaviour.CharacterMaster_OnInventoryChanged (On.RoR2.CharacterMaster+orig_OnInventoryChanged orig, RoR2.CharacterMaster self) (at D:/RoR2Related/Projects/MSUDev/Packages/teammoonstorm-moonstormsharedutils/Runtime/Code/Components/BaseItemMasterBehaviour.cs:196)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Hook<RoR2.CharacterMaster::OnInventoryChanged>?-1551338092(RoR2.CharacterMaster)
RoR2.Inventory.HandleInventoryChanged () (at <a149dbd7438a4e73b97d6600b3a2bbd4>:IL_000A)
RoR2.Inventory.SetEquipment (RoR2.EquipmentState equipmentState, System.UInt32 slot) (at <a149dbd7438a4e73b97d6600b3a2bbd4>:IL_002E)
(wrapper dynamic-method) RoR2.Inventory.DMD<RoR2.Inventory::UpdateEquipment>(RoR2.Inventory)
RoR2.Inventory.MyFixedUpdate (System.Single deltaTime) (at <a149dbd7438a4e73b97d6600b3a2bbd4>:IL_0007)
RoR2.Inventory.FixedUpdate () (at <a149dbd7438a4e73b97d6600b3a2bbd4>:IL_0006)
```

**Comment by Bubbet** [comment]
I've fixed the nre in 1.8.15, but as for what took your stuff away i think that was another factor. The nre happened because the inventory changed when the body didn't exist anymore. The nre by itself shouldn't have removed your items or equipment.

**Comment by acanthii** [comment]
oh weird, alright then. tysm! 🤍
